src/main/java/ui/NotificationController.java

```diff
@@ -34,6 +32,12 @@ void showNotification(Notification notification) {
                     }));
             notificationPaneTimer = new TickingTimer("Notification Timer", notification.getTimeoutDuration(),
                     integer -> {}, () -> Platform.runLater(this::triggerTimeoutAction), TimeUnit.SECONDS);
+            notificationPane.setOnHiding(event -> Platform.runLater(() -> {
```

Comment:
The hiding event seems to always originate from the UI thread, so the Platform.runLater isn't needed.

Can we factor the callback out into its method? It looks kind of similar to triggerTimeoutAction so maybe it could be merged.